runner/logger.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+type LogMessage struct {
+	Level   string `json:"level"`
+	Message string `json:"message"`
+	Type    string `json:"type"`
+}
+
+func (m LogMessage) toJsonString() string {
+	jsonStruct, err := json.Marshal(m)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(jsonStruct)
+}
+
+type CronRunnerLogger struct {
+	Format      string
+	Destination string
+	Type        string
+	logger      *log.Logger
+}
+
+func NewCronRunnerLogger(format, dest string) *CronRunnerLogger {
+	logOpts := 0
+
+	var logWriter logWriter
+	var logger *log.Logger
+
+	if dest == "os.Stdout" {
+		if format == "json" {
+			logWriter = NewLogWriter("cron-control", "cron-control-runner", false)
+		} else {
+			// original log format
+			logWriter = NewLogWriter("cron-control", "cron-control-runner", true)
+		}
+		logger = log.New(logWriter, "", logOpts)
+	} else {
+		fileWriter := NewFileWriter(dest, "cron-control", "cron-control-runner")
```

Comment:
How come we don't have the format check here?

Reply:
That's is a good point. I must have missed that. I'll add the format checks for that